pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -176,26 +177,49 @@ func (c *Reconciler) reconcilePA(ctx context.Context, rev *v1.Revision) error {
 		return fmt.Errorf("revision: %q does not own PodAutoscaler: %q", rev.Name, paName)
 	}
 
+	logger.Debugf("Observed PA Status=%#v", pa.Status)
+	rev.Status.PropagateAutoscalerStatus(&pa.Status)
+
 	// Perhaps tha PA spec changed underneath ourselves?
 	// We no longer require immutability, so need to reconcile PA each time.
 	tmpl := resources.MakePA(rev, deployment)
 	logger.Debugf("Desired PASpec: %#v", tmpl.Spec)
-	if !equality.Semantic.DeepEqual(tmpl.Spec, pa.Spec) {
-		diff, _ := kmp.SafeDiff(tmpl.Spec, pa.Spec) // Can't realistically fail on PASpec.
-		logger.Infof("PA %s needs reconciliation, diff(-want,+got):\n%s", pa.Name, diff)
-
+	if !equality.Semantic.DeepEqual(tmpl.Spec, pa.Spec) || !annotationsPresent(pa.Annotations, tmpl.Annotations) {
 		want := pa.DeepCopy()
 		want.Spec = tmpl.Spec
-		if pa, err = c.client.AutoscalingV1alpha1().PodAutoscalers(ns).Update(ctx, want, metav1.UpdateOptions{}); err != nil {
-			return fmt.Errorf("failed to update PA %q: %w", paName, err)
+		// copy template annotations over while preserving existing ones
+		maps.Copy(want.Annotations, tmpl.Annotations)
```

Comment:
This introduced a bug - when we want to remove the `min-scale` annotation this will preserve the KPA's min-scale. 

Thus there is no way to remove an annotation and have it propagate to a child resource